### ISSUE TITLE
using Rembulan as a Lua implementation?

### ISSUE THREAD

**Issue by keturn** [MovingBlocks/Kallisti#1]
I've just run across this as I'm testing something that involves Terasology's use of native libraries, and that includes JNLua.

I was wondering if Kallisti could use an implementation of Lua written for the JVM, avoiding the use of native implementations.

I discovered that [OpenComputers does fall back to LuaJ](https://ocdoc.cil.li/#persistence), but then the persistence features of Eris don't work, which sounds like a pretty big drawback.

I also found this about [Rembulan on OpenComputers](https://oc.cil.li/topic/1059-rembulan-a-new-lua-53-implementation-for-the-jvm-built-with-sandboxing-in-mind/) which sounded promising, but I don't know if anything came of it.

The [original Rembulan repository](https://github.com/mjanicek/rembulan) is dormant, but there are a few forks that have had activity to make updated builds.

* 2020: https://github.com/halset/rembulan/commits/master - looks like [Wizards of Lua](https://www.wizards-of-lua.net/) use Rembulan?
* 2018: https://github.com/JuKu/rembulan/commits/master

**Comment by asiekierka** [comment]
I'm not sure if it's worth the incompatibilities to have *two* slightly different implementations - and JNLua is based on the official Lua implementation. I think decent faith in Rembulan would be required to adopt it in place of JNLua/eris.

**Comment by keturn** [comment]
I agree that trying to keep up two implementations would be far too much work! You certainly would want to pick one and stick with it.

If one of your design goals is to work with existing libraries of Lua code from other sources (e.g. Lua that people have written for OpenComputers), I can see how compatibility would be a priority, which makes bindings to the canonical implementation a good choice.

Mostly my motivation to look at other lua implementations was driven by the fact that I've been working on things related to distribution, installation, and loading requirements, and requiring a "native" library involves different things than the rest of what's involved with distributing a Java application. But that is probably not something to worry about most of the time, I imagine lua must be pretty portable and well-behaved as such things go.
